test/bench.js

```diff
@@ -1,47 +1,80 @@
-var bench = require('nanobench')
-var fs = require('fs')
-var history = {}
+import {spawn} from 'child_process'
 
-function clear () {
-  for (var i in history) {
-    delete history[i]
-  }
-}
+import test from 'ava'
+import pify from 'pify'
+import tspan from 'time-span'
+
+import m from '..'
+
+async function create (pidno) {
+  const code = `
+    console.log(process.pid);
+    setInterval(function(){}, 1000); // Does nothing, but prevents exit
+  `
+  let count = 0
+  const childs = []
 
-function getProcessList (cb) {
-  fs.readdir('/proc', function (err, list) {
-    if (err) throw err
+  return new Promise((resolve, reject) => {
+    for (let i = 0; i < pidno; i++) {
+      const child = spawn('node', ['-e', code], {windowsHide: true})
+      childs.push(child)
 
-    cb(list.filter(function (v) {
-      return !isNaN(parseInt(v))
-    }))
+      child.stdout.on('data', function (childs) {
+        if (++count === pidno) resolve(childs)
+      }.bind(this, childs))
+      child.stderr.on('data', function (data) {
+        reject(data.toString())
+      })
+      child.on('error', reject)
+    }
   })
 }
 
-getProcessList(function (list) {
-  console.log('Benching %d process', list.length)
+async function destroy (childs) {
+  childs.forEach(child => child.kill())
+}
 
-  bench('procfile', function (b) {
-    var procfile = require('../lib/procfile')
-    b.start()
+async function execute (childs, pidno, times) {
+  var pids = childs.map(child => child.pid).slice(0, pidno)
 
-    procfile(list, {history: history}, function (err, data) {
-      if (err) throw err
+  const end = tspan()
+  try {
+    for (let i = 0; i < times; i++) {
+      await pify(m)(pids)
+    }
+    const time = end()
+    return Promise.resolve(time)
+  } catch (err) {
+    end()
+    return Promise.reject(err)
+  }
+}
```

Comment:
can we avoid using promises? Especially in benchmarks :p.

Reply:
The overhead is insignificant.
Without promises the code would be really too complicate.